classes/Events/SaveNotificationOnMessageSaveEvent.php

```diff
@@ -0,0 +1,195 @@
+                    'contexturlname'       => $message->contexturlname ?? null,
+                    'component_id'         => $this->getComponentId($message->component),
+                    'event_type_id'        => $eventTypeId,
+                    'notification_type_id' => 1,
+                    'should_email'         => (int) $shouldEmail
+                ]);
+            } catch (QueryException $e) {
+                return false;
+            }
+
+            if ($shouldEmail) {
+                $this->addToMailQueue($notification->id);
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Get event type id from string type. If it doesnt exist add it.
+     *
+     * @param mixed $eventType
+     * @return int
+     */
+    protected function getEventTypeId($eventType): int
+    {
+        $eventType = $eventType == null ? 'nullEventType' : $eventType;
+
+        $eventType = NotificationEventType::where(['name' => $eventType])->first();
+
+        if (!$eventType) {
+            $eventType = $this->createEventType($eventType);
+        }
+
+        return $eventType->id;
+    }
+
+    /**
+     * Get any enabled settings for specified event type against a user
+     *
+     * @param int $userId
+     * @param int $eventTypeId
+     * @return Collection
+     */
+    protected function notificationSettings(int $userId, int $eventTypeId): Collection
+    {
+        return NotificationSetting::where([
+            'user_id'       => $userId,
+            'event_type_id' => $eventTypeId,
+            'is_enabled'    => 1
+        ])->get();
+    }
+
+    /**
+     * Of enabled settings returned for event type, see if any are for email.
+     *
+     * @param Collection $settings
+     * @return bool
+     */
+    protected function shouldEmail(Collection $settings): bool
+    {
+        foreach ($settings as $setting) {
+            if ($setting->notification_type_id == 2) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * @param mixed $componentName
+     * @return int
+     */
+    protected function getComponentId($componentName): int
+    {
+        $component = NotificationComponent::where(['name' => $componentName])->first();
+
+        if (!$component) {
+            $component = $this->createComponentId($componentName);
+        }
+
+        return $component->id;
+    }
+
+    /**
+     * @param int $notificationId
+     * @return bool
+     */
+    protected function addToMailQueue(int $notificationId): bool
+    {
+        try {
+            NotificationEmailQueue::create([
+                'notification_id' => $notificationId,
+                'status' => 0
+            ]);
+            return true;
+        } catch (QueryException $e) {
+            return false;
+        }
+    }
+
+    /**
+     * @param string $eventType
+     * @return mixed
+     */
+    protected function createEventType(string $eventType)
+    {
+        try {
```

Comment:
Is it okay to have this try catch returning false? What error cases have we had which forced this, also will this potentially hide exceptions from showing where we might want them? Possibly not if this is just server-side, but if it is executed from the API, it may hide the issue from developers.

Reply:
Ah right yea i took out the original error logging as it kept breaking, ill have to add something back in.

Reply:
@aJamDonut The false statements have now been replaced with calls to the logger.